pallene/pallenelib.lua

```diff
@@ -0,0 +1,623 @@
+-- /*
```

Comment:
returning C code from a Lua code is a bit weird when viewing in a text editor 😅 
How would you feel if this was a regular old C file and we read it using `io` before returning the code?

Reply:
Yeah, it is certainly awkward. I have to manually switch the syntax highlighting color in my editor. The main problem is where to install the file, and how to find the path to the file after it is installed. I believe that with Luarocks you need to use a third-party library: https://github.com/hishamhm/datafile. I didn't want to add a dependency just so we could use this one file, but maybe it's clearer than the current hack... What do you think?

Reply:
I'm also in favor of having inline C code over adding a dependency.
But I also wonder why we can't just do `io.open("./pallene.h", "r")` to read the header.

Reply:
One of the things I want to achieve with this PR is to no longer care about the current working directory (issue #16). From now on, using relative paths is no longer allowed.